pkg/sources/sources.go

```diff
@@ -103,6 +103,30 @@ type SourceUnitEnumerator interface {
 	Enumerate(ctx context.Context, reporter UnitReporter) error
 }
 
+// BaseUnitReporter is a helper struct that implements the UnitReporter interface
+// and includes a JobProgress reference.
+type baseUnitReporter struct {
```

Comment:
What makes this a "base" unit reporter?

Reply:
the passed-in reporter gets wrapped into this, and so this reporter becomes the parent. 'base' made sense in that respect.